client/boot/common.js

```diff
@@ -116,6 +116,27 @@ const setupContextMiddleware = ( reduxStore ) => {
 	} );
 };
 
+function authorizePath() {
+	// TODO: flags=oauth should be present only in dev-like environments
+	// where we need to preserve the flag on full reloads and redirects
+	const redirectUri = new URL( '/api/oauth/token', window.location );
+	redirectUri.search = new URLSearchParams( {
+		flags: 'oauth',
```

Comment:
Feels so odd to hardcode that here. Should we wrap it in a check that does it only for the dev envs? DId you mean to do it as part of this PR?

Reply:
The `flags=oauth` query param was already hardcoded before this patch in the `getToken` function that stored the token and navigated to `/?flags=oauth`. This PR merely moves the code to a different location and adds the `next` query param that allows to configure the return route.

I plan to iterate further on this and eventually remove the query param. The goal is to preserve the `oauth` config flag on full reloads. And Jetpack Cloud already does a similar thing better: they have an `/oauth-override` endpoint that stores an OAuth-disabling flag to session storage and survives reloads of the tab: https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/jetpack/oauth-override.ts